presto-hive/src/test/java/io/prestosql/plugin/hive/s3/TestPrestoS3FileSystem.java

```diff
@@ -91,21 +94,31 @@ public void testStaticCredentials()
         Configuration config = new Configuration(false);
         config.set(S3_ACCESS_KEY, "test_access_key");
         config.set(S3_SECRET_KEY, "test_secret_key");
-        // the static credentials should be preferred
 
         try (PrestoS3FileSystem fs = new PrestoS3FileSystem()) {
-            verifyStaticCredentials(config, fs, "s3n://test-bucket/", "test_access_key", "test_secret_key");
+            AWSCredentials credentials = getStaticCredentials(config, fs, "s3n://test-bucket/");
+            assertEquals(credentials.getAWSAccessKeyId(), "test_access_key");
+            assertEquals(credentials.getAWSSecretKey(), "test_secret_key");
+            assertThat(credentials).isNotInstanceOf(AWSSessionCredentials.class);
+        }
+
+        config.set(S3_SESSION_TOKEN, "test_token");
+        try (PrestoS3FileSystem fs = new PrestoS3FileSystem()) {
+            AWSCredentials credentials = getStaticCredentials(config, fs, "s3n://test-bucket/");
+            assertEquals(credentials.getAWSAccessKeyId(), "test_access_key");
+            assertEquals(credentials.getAWSSecretKey(), "test_secret_key");
+            assertThat(credentials).isInstanceOfSatisfying(AWSSessionCredentials.class, sessionCredentials ->
```

Comment:
nit, move `AWSSessionCredentials.class, sessionCredentials ->` as new lines

Reply:
This seems worse:
```java
assertThat(credentials).isInstanceOfSatisfying(
        AWSSessionCredentials.class,
        sessionCredentials -> assertEquals(sessionCredentials.getSessionToken(), "test_token"));
```
With the above, the class name is close to the `isInstanceOf` call, and the `assertEquals` is by itself (easier to read than when preceded by a lambda argument).